src/client/components/modals/Organization.js

```diff
@@ -0,0 +1,244 @@
+          return true
+        } else {
+          throw new Error('Error checking membership')
+        }
+      })
+      .catch(err => {
+        dispatch(
+          requestedCheckMembershipError(ethereumAddress, {
+            message: err,
+          })
+        )
+      })
+  }
+
+  const validateAndSave = () => {
+    const errors = {}
+    let address = getFormValue('organizations', organizationId, 'address')
+
+    if (address.endsWith('.eth')) {
+      address = resolveEnsDomain(address)
+      if (!address) errors['organization'] = 'Could not resolve ENS address'
+    } else if (!validateDAOAddress(address))
+      errors['organization'] = 'Please provide valid DAO address'
+
+    if (isEmpty(errors)) {
+      checkMembershipAndSave(ethereumAddress, address)
+    } else {
+      setValidationErrors(errors)
+    }
+  }
+
+  const shortAddress = organizationId => {
+    let address = getFormValue('organizations', organizationId, 'address')
+    if (address.endsWith('.eth')) return address
+    return address.slice(0, 6) + '…' + address.slice(-4)
+  }
+  
+  const CheckWrapper = styled.div`
+    height: 146px;
+    display: flex;
+    align-items: center;
+    padding-bottom: 13px;
+    > :first-child {
+      margin-left: 10px;
+      margin-right: 30px;
+    }
+  `
+
+  const CheckStarted = () => (
+    <ModalWrapper>
+      <CheckWrapper>
+        <AnimationLoadingCircle />
+        <Text size="xxlarge">
+          Validating your membership to{' '}
+          {shortAddress(organizationId)}...
+        </Text>
+      </CheckWrapper>
+    </ModalWrapper>
+  )
+
+  const CheckSuccess = address => (
+    <ModalWrapper>
+      <CheckWrapper>
+        <AddOrganizationSuccess />
+        <Text size="xxlarge">
+          We found {shortAddress(organizationId)}{' '}
+          and confirmed you are a member
+        </Text>
+      </CheckWrapper>
+    </ModalWrapper>
+  )
+
+  const CheckError = () => (
+    <ModalWrapper>
+      <CheckWrapper>
+        <AddOrganizationError />
+        <div
+          css={`
+            height: 100%;
+            position: relative;
+            display: flex;
+            align-items: center;
+          `}
+        >
+          <Text.Block size="xxlarge">
+            We could not verify your membership to{' '}
+            {shortAddress(organizationId)}
+          </Text.Block>
+
+          <Text.Block
+            css={`
+              text-align: right;
+              margin-bottom: 0px;
+              text-decoration: underline;
+              cursor: pointer;
+              vertical-align: text-bottom;
+              position: absolute;
+              bottom: 0;
+              right: 0;
+            `}
+            color={theme.accent}
+            onClick={() =>
+              dispatch(requestedCheckMembershipClean(ethereumAddress))
+            }
+          >
+            Back to previous screen
+          </Text.Block>
+        </div>
+      </CheckWrapper>
+    </ModalWrapper>
+  )
+
+  if (
```

Comment:
This is the right idea, but if for some reason two of these are true (the state shows a bug) the UI will look really awkward. We can shorten this a lot with just a bunch of if/else if/else statemnets:

```js
if (checkingMembership) return <CheckStarted />
else if (checkedMembershipSuccess) return <CheckSuccess />
else if (checkedMembershipError) return <CheckError />
else return <ModalWrapper>{...}</ModalWrapper>
```

Reply:
Fixed.